darwin/dataset/remote_dataset_v2.py

```diff
@@ -842,3 +862,136 @@ def register_multi_slotted(
+    """
+    multi_file_items, local_files = [], []
+    for directory in search_files:
+        files_in_directory = list(
+            find_files(
+                [directory],
+                files_to_exclude=files_to_exclude,
+                recursive=False,
+                sort=True,
+            )
+        )
+        if not files_in_directory:
+            print(
+                f"Warning: There are no files in the first level of {directory}, skipping directory"
+            )
+            continue
+        multi_file_item = MultiFileItem(
+            Path(directory), files_in_directory, ItemMergeMode(item_merge_mode), fps
+        )
+        multi_file_items.append(multi_file_item)
+        local_files.extend(multi_file_item.files)
+
+    if not multi_file_items:
+        raise ValueError(
+            "No valid folders to upload after searching the passed directories for files"
+        )
+    return local_files, multi_file_items
+
+
+def _find_files_to_upload_no_merging(
+    search_files: List[PathLike],
+    files_to_exclude: List[PathLike],
+    path: Optional[str],
+    fps: int,
+    as_frames: bool,
+    extract_views: bool,
+    preserve_folders: bool,
+    uploading_files: List[LocalFile],
+) -> List[LocalFile]:
+    """
+    Finds files to upload as single-slotted dataset items. Recursively searches the passed directories for files.
+
+    Parameters
+    ----------
+    search_files : List[PathLike]
+        List of directories to search for files.
+    files_to_exclude : Optional[List[PathLike]]
+        List of files to exclude from the file scan.
+    path : Optional[str]
+        Path to store the files in.
+    fps: int
+        When uploading video files, specify the framerate.
+    as_frames: bool
+        When uploading video files, specify whether to upload as a list of frames.
+    extract_views: bool
+        When uploading volume files, specify whether to split into orthogonal views.
+    preserve_folders: bool
+        Specify whether or not to preserve folder paths when uploading.
+    uploading_files : List[LocalFile]
+        List of files to upload.
+
+    Returns
+    -------
+    List[LocalFile]
+        List of files to upload.
+    """
+    generic_parameters_specified = (
+        path is not None or fps != 0 or as_frames is not False
+    )
+    if uploading_files and generic_parameters_specified:
+        raise ValueError("Cannot specify a path when uploading a LocalFile object.")
+
+    for found_file in find_files(search_files, files_to_exclude=files_to_exclude):
+        local_path = path
+        if preserve_folders:
+            source_files = [
+                source_file
+                for source_file in search_files
+                if is_relative_to(found_file, source_file)
+            ]
+            if source_files:
+                local_path = str(
+                    found_file.relative_to(source_files[0]).parent.as_posix()
+                )
```

Comment:
This is the old current `push` logic that I placed unchanged into a function to separate it from multi-file `push`. `is_relative_to()` is basically saying: Does the 2nd path sit somewhere inside the 1st? If not, we can't push it with `preserve_folders` because there would be no difference between the two paths (which is what `preserve_folders` represents in the dataset)

The reason this list comprehension is necessary is that it's possible for `search_files` to contain multiple directories, and we need to select the correct source_file directory to calculate the relative path from. For example:

If `search_files` contains two directories: `dir1` and `dir2`. If found_file is `dir2/subdir/file.png`, the list comprehension ensures that the relative path is calculated with respect to `dir2`, not `dir1`